File: internal/domain/audit_log.go
package domain

import (
	"encoding/json"
	"time"
)

type AuditLog struct {
	ID           int64           `json:"id"`
	UserID       *int64          `json:"user_id"`
	Action       string          `json:"action"`
	ResourceType *string         `json:"resource_type"`
	ResourceID   *int64          `json:"resource_id"`
	IPAddress    *string         `json:"ip_address"`
	UserAgent    *string         `json:"user_agent"`
	Details      json.RawMessage `json:"details"`
	CreatedAt    *time.Time      `json:"created_at"`
}

type CreateAuditLogAction struct {
	UserID       *int64
	Action       string
	ResourceType *string
	ResourceID   *int64
	IPAddress    *string
	UserAgent    *string
	Details      json.RawMessage
}

// Common audit actions
const (
	AuditActionUserLogin          = "user_login"
	AuditActionUserLogout         = "user_logout"
	AuditActionUserRegister       = "user_register"
	AuditActionUserUpdate         = "user_update"
	AuditActionUserDeactivate     = "user_deactivate"
	AuditActionUserActivate       = "user_activate"
	AuditActionPasswordChange     = "password_change"
	AuditActionPasswordReset      = "password_reset"
	AuditActionEmailVerify        = "email_verify"
	AuditActionEmailResend        = "email_resend"
	AuditActionSessionCreate      = "session_create"
	AuditActionSessionRevoke      = "session_revoke"
	AuditActionSessionRevokeAll   = "session_revoke_all"
	AuditActionAccountLockout     = "account_lockout"
	AuditActionSuspiciousActivity = "suspicious_activity"
	AuditActionDataExport         = "data_export"
	AuditActionPrivacySettings    = "privacy_settings"
)

// Common resource types
const (
	AuditResourceTypeUser     = "user"
	AuditResourceTypeSession  = "session"
	AuditResourceTypePassword = "password"
	AuditResourceTypeEmail    = "email"
	AuditResourceTypeAccount  = "account"
	AuditResourceTypeData     = "data"
	AuditResourceTypePrivacy  = "privacy"
	AuditResourceTypeDevice   = "device"
)

File: frontend/src/types/models/audit_log.ts
export type AuditLog = {
	id: number
	user_id: number | null
	action: string
	resource_type: string | null
	resource_id: number | null
	ip_address: string | null
	user_agent: string | null
	details: Record<string, any> | null
	created_at: string | null
}

// Common audit actions
export const AuditActions = {
	USER_LOGIN: 'user_login',
	USER_LOGOUT: 'user_logout',
	USER_REGISTER: 'user_register',
	USER_UPDATE: 'user_update',
	USER_DEACTIVATE: 'user_deactivate',
	USER_ACTIVATE: 'user_activate',
	PASSWORD_CHANGE: 'password_change',
	PASSWORD_RESET: 'password_reset',
	EMAIL_VERIFY: 'email_verify',
	EMAIL_RESEND: 'email_resend',
	SESSION_CREATE: 'session_create',
	SESSION_REVOKE: 'session_revoke',
	SESSION_REVOKE_ALL: 'session_revoke_all',
	ACCOUNT_LOCKOUT: 'account_lockout',
	SUSPICIOUS_ACTIVITY: 'suspicious_activity',
	DATA_EXPORT: 'data_export',
	PRIVACY_SETTINGS: 'privacy_settings',
} as const

// Common resource types
export const AuditResourceTypes = {
	USER: 'user',
	SESSION: 'session',
	PASSWORD: 'password',
	EMAIL: 'email',
	ACCOUNT: 'account',
	DATA: 'data',
	PRIVACY: 'privacy',
	DEVICE: 'device',
} as const
